policy-controller/k8s/index/src/index.rs

```diff
@@ -201,15 +216,192 @@ impl Index {
 
     fn apply_route<R>(&mut self, route: R)
     where
-        R: ResourceExt,
+        R: ResourceExt + GetParentRefs + Clone + IntoRouteBinding,
         InboundRouteBinding: TryFrom<R>,
         <InboundRouteBinding as TryFrom<R>>::Error: std::fmt::Display,
     {
+        fn make_patch(name: &str, status: gateway::HttpRouteStatus) -> serde_json::Value {
+            serde_json::json!({
+                "apiVersion": "policy.linkerd.io/v1alpha1",
+                "kind": "HTTPRoute",
+                "name": name,
+                "status": status,
+            })
+        }
+
         let ns = route.namespace().expect("HttpRoute must have a namespace");
         let name = route.name_unchecked();
         let _span = info_span!("apply", %ns, %name).entered();
 
-        let route_binding = match route.try_into() {
+        let mut skip_index = false;
+        let mut parent_refs = vec![];
+        let mut parent_statuses: Vec<gateway::RouteParentStatus> = vec![];
+        for parent_ref in route.clone().get_parent_refs() {
+            match parent_ref {
+                Ok(parent) => {
+                    let InboundParentRef::Server(ref name) = parent;
+
+                    // Ensure the Server exists.
+                    if self
+                        .namespaces
+                        .by_ns
+                        .get(&ns)
+                        .and_then(|ns| ns.policy.servers.get(name))
+                        .is_none()
+                    {
+                        skip_index = true;
+                        parent_statuses.push(gateway::RouteParentStatus {
+                            parent_ref: gateway::ParentReference {
+                                group: Some("policy.linkerd.io".to_string()),
+                                kind: Some("Server".to_string()),
+                                namespace: Some(ns.clone()),
+                                name: name.to_string(),
+                                section_name: None,
+                                port: None,
+                            },
+                            controller_name: "policy.linkerd.io/status-controller".to_string(),
+                            conditions: vec![k8s::Condition {
+                                last_transition_time: k8s::Time(Utc::now()),
+                                message: "".to_string(),
+                                observed_generation: None,
+                                reason: "NoMatchingParent".to_string(),
+                                status: "False".to_string(),
+                                type_: "Accepted".to_string(),
+                            }],
+                        })
```

Comment:
it might be nice to have a helper function that constructs a `RouteParentStatus` given a `Vec<Condition>` and a namespace/name, since we repeat a bunch of the same boilerplate, like the kind, group, and controller name a bunch of times in this method?